jdaviz/configs/specviz/plugins/viewers.py

```diff
@@ -373,15 +373,9 @@ def add_data(self, data, color=None, alpha=None, **layer_state):
             try:
                 uc.to_unit(data, data.find_component_id("flux"), [1, 1],
                            u.Unit(self.state.y_display_unit))  # Error if incompatible
-            except Exception as err:
-                # Raising exception here introduces a dirty state that messes up next load_data
-                # but not raising exception also causes weird behavior unless we remove the data
-                # completely.
-                self.session.hub.broadcast(SnackbarMessage(
-                    f"Failed to load {data.label}, so removed it: {repr(err)}",
-                    sender=self, color='error'))
-                self.jdaviz_app.data_collection.remove(data)
-                return False
+            except Exception:
+                pass
+
```

Comment:
this is just for testing loading in incompatible units into the same viewer.  We would eventually want to allow loading these incompatible units (surface brightness vs flux, for example) into the app, but not into the same viewer.  The data-menu logic would then control what data can be available to be loaded into a given viewer.